### ISSUE TITLE
Some Feature Ideas and bugs for this client

### ISSUE THREAD

**Issue by SirNapkin1334** [yourWaifu/Unofficial-Discord-3DS-Client#3]
Hello. I've been using and testing this client for about a day, and I really like it. It's a good example of how a program should be made. However, despite this, there are some problems, and I also have some feature ideas.
First of all, it appears to crash on almost a schedule. It crashes to home menu about every five minutes, so I have to re-start the program.
Now, some ideas. I would like it to automatically display some history, rather than having to press Y. It'd be nice to have a selector GUI on the bottom screen that allows you to tap, or at least see, the channels. Besides that, great project! 
![grog](https://user-images.githubusercontent.com/31591796/31918861-a32a55b4-b813-11e7-93f9-2188756df158.gif)

**Comment by yourWaifu** [comment]
The random crashing doesn't seem to be affecting me, and plus it happens randomly, so this crash is going to be a tough one to fix. About the feature request, before a GUI like that can be made, a few things will need to be implemented to get that to work. Things like drawing and loading web image formats for it look the way I want it to.

**Comment by SirNapkin1334** [comment]
Yes, I had thought of the loading server images. Also, I thought of some new key functions:
Dpad Up/Down = Scroll
A = send message
B = delete last sent message/select message to delete
X = edit last sent message/select message to edit
Y + Dpad Up/Down = Fast Scroll
L/R = Switch Server
Dpad Left/Right = Switch Channel
Select = Toggle Debug Mode (show bytes sent, etc., what is currently shown on lower screen.

**Comment by yourWaifu** [comment]
I like most of what you are subjecting, but I would change
```
Switch Channel = Dpad Up/Down
Switch Server  = Dpad Left/Right
Select         = Touchscreen Tap     # messages will also be on the bottom screen
Scroll         = Touchscreen Swipe   # like on a smartphone
Delete or Edit = Swipe Left or Right or Long Press on Touchscreen
```
My plans are a bit harder to implement then what you are asking tho, I'll consider your as the first version.
